### ISSUE TITLE
Error: table id is incorrect for snowflake_table_constraint

### ISSUE THREAD

**Issue by freylafert** [Snowflake-Labs/terraform-provider-snowflake#2535]
### Terraform CLI and Provider Versions

terraform --version
Terraform v1.5.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v5.37.0
+ provider registry.terraform.io/hashicorp/null v3.2.2
+ provider registry.terraform.io/snowflake-labs/snowflake v0.86.0

### Terraform Configuration

```terraform
resource "snowflake_table_constraint" "primary_key" {
  depends_on = [snowflake_table.table]
  provider   = snowflake.account_admin
  for_each   = var.tables
  name       = "${each.value.name}_PRIMARY_KEY"
  type       = "PRIMARY KEY"
  table_id   = snowflake_table.table[each.key].id
  columns    = each.value.primary_key_fields
}
```

### Expected Behavior

apply with no errors

### Actual Behavior

╷
│ Error: table id is incorrect: {CNT_DEV|OLEG_CNT_DEV|IP_TO_NAMED_LOCATION}
│ 
│   with module.snowflake.snowflake_table_constraint.primary_key["ip_to_named_location"],
│   on ../reporting/snowflake/snowflake.tf line 134, in resource "snowflake_table_constraint" "primary_key":
│  134: resource "snowflake_table_constraint" "primary_key" {
│ 
╵

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

High

### Additional Information

it looks like in version 0.86.0 we need to pass table_id = snowflake_table.t.qualified_name vs table_id = snowflake_table.t.id in the previous versions. While it looks like for existing tables there was no issues to stay with snowflake_table.t.id it fails for new tables apply failes with the error mentioned above. When I change tf configuration to use qualified_name instead of id, I got this error for all existing tables. So it looks like I am stuck now.

**Comment by sfc-gh-asawicki** [comment]
Hey @freylafert. Thanks for reaching out to us.

This is apparently an oversight on our side and it should be added to the migration notes.

Identifiers are a never-ending topic. This is why we will soon rework them (https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#our-roadmap). The problem with this resource was that it was using the deprecated identifier parser. Also, the docs were inconsistent because the example showed using the `table.id` but the description of the `table_id` parameter never allowed such a value (`table.id` is a `|` delimited identifier representation and only the `.` separated values were listed in the docs: https://registry.terraform.io/providers/Snowflake-Labs/snowflake/0.85.0/docs/resources/table_constraint#required - as I can see there is some problem with rendering the proper values here, I will also correct that).

 I think that the easiest way to go from here would be to use the existing [resource migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/5e2dbf301815ad7fe5215ba9c45b1ec4b7b51c16/docs/technical-documentation/resource_migration.md) to remove from state and reimport the affected resources - I will add this to the migration notes.

We will provide a temporary workaround in the next version (next week) so that the manual migration is not necessary, so you can wait for the next version which should solve the problem.

**Comment by freylafert** [comment]
Thank you, that makes sense. It will be fixed in the version 0.87.0, right?

**Comment by sfc-gh-asawicki** [comment]
In 0.87.0 or 0.88.0. Preferably, in 0.87.0 but I can't promise that.

**Comment by rjminchuk** [comment]
https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2553 Just opened this.. will close it. Having the same problem. Thanks.

**Comment by philippeboyd** [comment]
@sfc-gh-asawicki can we also get rid of the emojis in resource IDs? I though we already [had this conversation](https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1541)

![image](https://github.com/Snowflake-Labs/terraform-provider-snowflake/assets/3239656/4be74ae6-c5ae-4bc5-a406-ee2120e3fca1)

**Comment by sfc-gh-asawicki** [comment]
Hey @philippeboyd, you never had a conversation with me.

This is one of the last places with the ❄️-separated identifier. We will get rid of all of them with the upcoming resources redesign: https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#our-roadmap.

Also, please submit new tickets for different problems instead of adding them to the existing unrelated issues.

**Comment by sfc-gh-asawicki** [comment]
Hey @freylafert. We have released the fix as part of [v0.87.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.87.0) release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/v0.87.0/MIGRATION_GUIDE.md#v0860--v0870) during the update. Please confirm that the issue is resolved in the newest version. Thanks!
